hydra-node/src/Hydra/Snapshot.hs

```diff
@@ -78,10 +78,9 @@ instance forall tx. IsTx tx => SignableRepresentation (Snapshot tx) where
     LBS.toStrict $
       serialise (toData $ toBuiltin $ serialiseToRawBytes headId)
         <> serialise (toData $ toBuiltin $ toInteger number) -- CBOR(I(integer))
-        <> serialise (toData $ toBuiltin $ hashUTxO @tx utxo) -- CBOR(B(bytestring)
-        -- TODO: is serializing separately fine or we should get the diff
-        -- between two utxos and sign that? In the spec we have added new field
-        <> maybe mempty (serialise . (toData . toBuiltin . hashUTxO @tx)) utxoToDecommit -- CBOR(B(bytestring)
+        -- TODO: In the spec we have added new field but here we sign the diff
+        -- between two utxos.
+        <> serialise (toData $ toBuiltin $ hashUTxO @tx (utxo `withoutUTxO` maybe mempty id utxoToDecommit)) -- CBOR(B(bytestring)
```

Comment:
Must: keep the serialization as it was before, i.e. hashing the `utxoToDecommit`